DuckDuckGo/LocalNotificationsLogic.swift

```diff
@@ -0,0 +1,256 @@
+//
+//  LocalNotificationsLogic.swift
+//  DuckDuckGo
+//
+//  Copyright © 2019 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import UIKit
+import UserNotifications
+
+protocol NotificationsStore {
+    
+    var notificationsEnabled: Bool { get set }
+    
+    func scheduleStatus(for notification: LocalNotificationsLogic.Notification) -> LocalNotificationsLogic.ScheduleStatus?
+    
+    func didSchedule(notification: LocalNotificationsLogic.Notification, date: Date)
+    func didFire(notification: LocalNotificationsLogic.Notification)
+    
+    func didCancel(notification: LocalNotificationsLogic.Notification)
+
+}
+
+protocol LocalNotificationsLogicDelegate: class {
+    
+    func displayHomeHowInstructions(for: LocalNotificationsLogic)
+}
+
+// swiftlint:disable nesting
```

Comment:
This could be disabled / enabled around the nested type rather than the whole thing